api/models/reviews.js

```diff
@@ -1,28 +1,44 @@
 const { response } = require("express");
 const db = require("../../db/connection");
 
-exports.selectReviews = () => {
-  return db
-    .query(
-      `SELECT 
-    title,
-    designer,
-    owner,
-    reviews.review_id,
-    review_img_url ,
-    category,
-    reviews.created_at,
-    reviews.votes,
-   (SELECT COUNT(comment_id) 
-    FROM comments
-    WHERE review_id = reviews.review_id) AS comment_count FROM reviews
-    LEFT JOIN comments ON reviews.review_id = comments.review_id
-    GROUP BY reviews.review_id
-    ORDER BY reviews.created_at DESC`
-    )
-    .then((reviews) => {
-      return reviews.rows;
-    });
+exports.selectReviews = (category, sort_by, order) => {
+  const querys = [];
+  if (sort_by !== undefined) {
+    if (["title", "designer", "owner", "category", "votes"].includes(sort_by)) {
+      sort_by = sort_by;
+    }
+  } else {
+    sort_by = `reviews.created_at`;
+  }
+  let reviews = `SELECT 
+  title,
+  designer,
+  owner,
+  reviews.review_id,
+  review_img_url ,
+  category,
+  reviews.created_at,
+  reviews.votes,
+  (SELECT COUNT(comment_id) 
+  FROM comments
+  WHERE review_id = reviews.review_id) AS comment_count FROM reviews
+  LEFT JOIN comments ON reviews.review_id = comments.review_id
+  `;
+  let orderByText = `GROUP BY reviews.review_id 
+  ORDER BY ${sort_by}`;
+  if (category !== undefined) {
+    reviews += ` WHERE category = $1`;
+    querys.push(category);
+  }
+  if (order === "asc") {
+    reviews += orderByText += ` ASC`;
+  } else {
+    reviews += orderByText += ` DESC`;
+  }
```

Comment:
If you set a default, then you dont need this. 
It will either be the default value - if none set
or the other one they pass in. 
You can just use the `order` variable